### ISSUE TITLE
make it possible to announce label automatically with screenreader when focusing in window

### ISSUE THREAD

**Issue by beqabeqa473** [dotnet/winforms#10652]
### .NET version

8.0

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

I have a form, which has one label, a progressbar and a cancel button.
When i am showing a from from another form, a title of window is announced and cancel button.
even if i show form with showDialog same thing occures.

### Steps to reproduce

create form described in issue description.

**Comment by merriemcgaw** [comment]
Usually this is handled by the verbosity settings in Narrator if someone wants to see/announce the labels https://support.microsoft.com/en-us/windows/chapter-7-customizing-narrator-ce950246-c915-0d44-9be6-fb474387a285
If you want to notify the progress changes that is another scenario and requires you to create Notifications. 
@Olina-Zhang can your team make sure the scenario described above is working properly in .NET 8 and nothing weird has happened?

**Comment by Zheng-Li01** [comment]
@beqabeqa473  @merriemcgaw 

Verified the issue on both .NET Framework 4.8.1 and .NET 8.0/9.0, which have the same results.

When the Narrator is Scan Off mode, the user can access the non-focusable controls (eg. label, progressbar) by using "CapsLock + keyboard arrows" 

When the Narrator is Scan On mode, the user can access the non-focusable controls (eg. label, progressbar) by using  keyboard arrows.

Note:
Using the  "CapsLock + Space" to enable/disable the "Scan mode"

**Framework:**
![FrameworkBehaviour](https://github.com/dotnet/winforms/assets/38325459/4cc24c56-3c74-4548-bbbb-e18a652aa300)

**Core:**
![CoreBehaviour](https://github.com/dotnet/winforms/assets/38325459/d4f821b5-c9ec-49d9-9247-a3c33a41777c)

**Comment by beqabeqa473** [comment]
no, it is verifyable with nvda that static text is not read by screenreader.

On 1/17/24, v-zhgl ***@***.***> wrote:
> @beqabeqa473  @merriemcgaw
>
> Verified the issue on both .NET Framework 4.8.1 and .NET 8.0/9.0, which have
> the same results.
>
> When the Narrator is Scan Off mode, the user can access the non-focusable
> controls (eg. label, progressbar) by using "CapsLock + keyboard arrows"
>
> When the Narrator is Scan On mode, the user can access the non-focusable
> controls (eg. label, progressbar) by using  keyboard arrows.
>
> Note:
> Using the  "CapsLock + Space" to enable/disable the "Scan mode"
>
> **Framework:**
> ![FrameworkBehaviour](https://github.com/dotnet/winforms/assets/38325459/4cc24c56-3c74-4548-bbbb-e18a652aa300)
>
> **Core:**
> ![CoreBehaviour](https://github.com/dotnet/winforms/assets/38325459/d4f821b5-c9ec-49d9-9247-a3c33a41777c)
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/dotnet/winforms/issues/10652#issuecomment-1895146783
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by Zheng-Li01** [comment]
@beqabeqa473, thanks for your quickly response. 

The nvda can announce the static text of label when focusing it by using mouse. However cannot find the way, such as key combination or hot key to focus on label by using keyboard, do you have any ideals for this?
![Nvda](https://github.com/dotnet/winforms/assets/38325459/6bce2377-e78a-40e9-b587-57adfa1a30e8)

**Comment by beqabeqa473** [comment]
as it works with an usual winapi win32 app, static text should be
announced automatically when focusing a windows, but with winforms,
this doesn't happen.

On 1/17/24, v-zhgl ***@***.***> wrote:
> @beqabeqa473, thanks for your quickly response.
>
> The nvda can announce the static text of label when focusing it by using
> mouse. However cannot find the way, such as key combination or hot key to
> focus on label by using keyboard, do you have any ideals for this?
> ![Nvda](https://github.com/dotnet/winforms/assets/38325459/6bce2377-e78a-40e9-b587-57adfa1a30e8)
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/dotnet/winforms/issues/10652#issuecomment-1895266092
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by Zheng-Li01** [comment]
@beqabeqa473, thanks for your update.

Following your description, test the issue no both Narrator and NVDA with Show()/ShowDialog methods. see below details..
**Setting the AccessibleName for label.**
When calling the Show() method, the Narrator and NVDA not announce the AccessibleName for label
When calling the ShowDialog() method, the Narrator and NVDA can announce the AccessibleName for label

Narrator:
![NarratorBehaviour](https://github.com/dotnet/winforms/assets/38325459/ab7bfdff-ba26-40ec-bc23-df60aea213c1)
NVDA:
![NVDABehaviour](https://github.com/dotnet/winforms/assets/38325459/6a2e9cda-8e7d-4465-a624-06b047e84c5c)

**Not setting the AccessibleName for label.**
When calling the ShowDialog()/Show() methods, the Narrator and NVDA not announce the static text for label.
**Narrator:**
![NarratorWithoutAccessibilityName](https://github.com/dotnet/winforms/assets/38325459/3bd7d66d-1f33-45cd-ad6f-cf4f3d1c71f1)
**NVDA:**
![NVDAWithoutAccessibilityName](https://github.com/dotnet/winforms/assets/38325459/cfba7149-fb73-476f-8a08-eb9a26aee32c)

When calling the show() method, the Form2 type is **Window**; and calling the showdialog() method, the Form2 type is "Dialog".
cc/: @merriemcgaw

**Comment by Olina-Zhang** [comment]
We have got the information that Screen readers read content of dialogs and regular windows differently. They read multiple controls for dialogs but only title and the focused control for regular windows. About announcing in modal windows described here: https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-automation-element-propids (see UIA_IsDialogPropertyId). 
![image](https://github.com/dotnet/winforms/assets/26474449/1a74faac-8887-4a0e-8aa7-53bd14d3c093)

**Comment by beqabeqa473** [comment]
anyway, this type of possibility is not opened up for developers to
make it possible to mark a form as a dialog for screenreader and only
for screenreader.
And even that, only label with accessible name is read by the
screenreader what in my opinion is wrong.

On 1/17/24, v-olzhan ***@***.***> wrote:
> We have got the information that Screen readers read content of dialogs and
> regular windows differently. They read multiple controls for dialogs but
> only title and the focused control for regular windows. About announcing in
> modal windows described here:
> https://docs.microsoft.com/en-us/windows/win32/winauto/uiauto-automation-element-propids
> (see UIA_IsDialogPropertyId).
> ![image](https://github.com/dotnet/winforms/assets/26474449/1a74faac-8887-4a0e-8aa7-53bd14d3c093)
>
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/dotnet/winforms/issues/10652#issuecomment-1895510337
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
with best regards Beqa Gozalishvili
Tell: +995593454005
Email: ***@***.***
Web: https://gozaltech.org
Skype: beqabeqa473
Telegram: https://t.me/gozaltech
facebook: https://facebook.com/gozaltech
twitter: https://twitter.com/beqabeqa473
Instagram: https://instagram.com/beqa.gozalishvili

**Comment by Tanya-Solyanik** [comment]
@beqabeqa473  - you could try adding additional information, contained in the label, to the Form accessible name or description. We can't change the documented behavior that other UI frameworks are following, it would be unexpected for the users.
